### ISSUE TITLE
Movies stuck in Activity, not importing successfully

### ISSUE THREAD

**Issue by nightah** [Radarr/Radarr#2501]
**Description:**

I'm having an issue where all Movies get stuck in the Activity tab due to what appears to be a race condition.
I think may have something to do with the Automatic Folder Renaming and the Movie Folder Format `{Movie Title} [{Release Year}]  [{Quality Title}]` options when importing.

Effectively the Movie gets downloaded, and moved appropriately into the final location, however, if you check the movie, Radarr doesn't seem to notice that the file is there so it's still marked as missing.

The initial folder that the movie gets imported into doesn't include the format so the initial folder location it's looking at is `/downloads/Movies/Serendipity [2001] []` however upon the import it creates the folder `/downloads/Movies/Serendipity [2001] [Bluray-1080p]` and moves the file appropriately.

Then Radarr errors saying it couldn't import the movie, logs below.

**Radarr Version:**
0.2.0.935 (Using the latest linuxserver.io Docker container)
**Mono Version:**
5.8.0.108
**Logs:**
```
8-2-10 23:32:52.4|Debug|DiskScanService|Scanning '/downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/' for video files
18-2-10 23:32:52.4|Debug|DiskScanService|1 video files were found in /downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/
18-2-10 23:32:52.4|Debug|ImportDecisionMaker|Analyzing 1/1 files.
18-2-10 23:32:52.4|Debug|QualityParser|Trying to parse quality for /downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/Serendipity.2001.1080p.BluRay.X264-ml.mkv
18-2-10 23:32:52.4|Debug|QualityParser|Trying to parse quality for Serendipity.2001.1080p.BluRay.X264-ml.mkv
18-2-10 23:32:52.4|Debug|ImportDecisionMaker|Size: 7037285202
18-2-10 23:32:52.4|Debug|VideoFileInfoReader|Getting media info from /downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/Serendipity.2001.1080p.BluRay.X264-ml.mkv
18-2-10 23:32:52.4|Debug|GrabbedReleaseQualitySpecification|No download client item provided, skipping.
18-2-10 23:32:52.4|Debug|Parser|Parsing string 'Serendipity.2001.1080p.BluRay.x264-ML'
18-2-10 23:32:52.4|Debug|Parser|Episode Parsed. Serendipity - Season 01
18-2-10 23:32:52.4|Debug|Parser|Language parsed: English
18-2-10 23:32:52.4|Debug|QualityParser|Trying to parse quality for Serendipity.2001.1080p.BluRay.x264-ML
18-2-10 23:32:52.4|Debug|Parser|Quality parsed: Bluray-1080p v1
18-2-10 23:32:52.4|Debug|Parser|Release Group parsed: ML
18-2-10 23:32:52.4|Debug|VideoFileInfoReader|Getting media info from /downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/Serendipity.2001.1080p.BluRay.X264-ml.mkv
18-2-10 23:32:52.4|Debug|DetectSample|Runtime is over 90 seconds
18-2-10 23:32:52.4|Debug|ImportApprovedMovie|Decisions: 1
18-2-10 23:32:52.4|Debug|Parser|Parsing string 'Serendipity.2001.1080p.BluRay.x264-ML'
18-2-10 23:32:52.4|Debug|Parser|Movie Parsed. Serendipity - 2001
18-2-10 23:32:52.4|Debug|Parser|Language parsed: English
18-2-10 23:32:52.4|Debug|QualityParser|Trying to parse quality for Serendipity.2001.1080p.BluRay.x264-ML
18-2-10 23:32:52.4|Debug|Parser|Quality parsed: Bluray-1080p v1
18-2-10 23:32:52.4|Debug|Parser|Release Group parsed: ML
18-2-10 23:32:52.5|Debug|DiskProvider|Setting permissions: 0775 on /downloads/Movies/Serendipity [2001] [Bluray-1080p]
18-2-10 23:32:52.5|Debug|MovieFileMovingService|Moving movie file: /downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/Serendipity.2001.1080p.BluRay.X264-ml.mkv to /downloads/Movies/Serendipity [2001] [Bluray-1080p]/Serendipity.mkv
18-2-10 23:32:52.5|Debug|DiskTransferService|Move [/downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/Serendipity.2001.1080p.BluRay.X264-ml.mkv] > [/downloads/Movies/Serendipity [2001] [Bluray-1080p]/Serendipity.mkv]
18-2-10 23:32:52.5|Debug|DiskProvider|Setting permissions: 0775 on /downloads/Movies/Serendipity [2001] [Bluray-1080p]/Serendipity.mkv
18-2-10 23:32:52.5|Warn|ImportApprovedMovie|Couldn't import movie /downloads/Movies/Serendipity.2001.1080p.BluRay.x264-ML/Serendipity.2001.1080p.BluRay.X264-ml.mkv

[v0.2.0.935] System.IO.DirectoryNotFoundException: Could not find a part of the path '/downloads/Movies/Serendipity [2001] []'.
  at System.IO.__Error.WinIOError (System.Int32 errorCode, System.String maybeFullPath) [0x000f7] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.FileSystemEnumerableIterator`1[TSource].HandleError (System.Int32 hr, System.String path) [0x00006] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.FileSystemEnumerableIterator`1[TSource].CommonInit () [0x00054] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.FileSystemEnumerableIterator`1[TSource]..ctor (System.String path, System.String originalUserPath, System.String searchPattern, System.IO.SearchOption searchOption, System.IO.SearchResultHandler`1[TSource] resultHandler, System.Boolean checkHost) [0x000d6] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.FileSystemEnumerableFactory.CreateFileNameIterator (System.String path, System.String originalUserPath, System.String searchPattern, System.Boolean includeFiles, System.Boolean includeDirs, System.IO.SearchOption searchOption, System.Boolean checkHost) [0x00009] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.Directory.InternalGetFileDirectoryNames (System.String path, System.String userPathOriginal, System.String searchPattern, System.Boolean includeFiles, System.Boolean includeDirs, System.IO.SearchOption searchOption, System.Boolean checkHost) [0x00000] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.Directory.InternalGetFiles (System.String path, System.String searchPattern, System.IO.SearchOption searchOption) [0x00000] in <2e7c1c96edae44d496118948ca617c11>:0
  at System.IO.Directory.GetFiles (System.String path, System.String searchPattern, System.IO.SearchOption searchOption) [0x00038] in <2e7c1c96edae44d496118948ca617c11>:0
  at NzbDrone.Common.Disk.DiskProviderBase.GetFiles (System.String path, System.IO.SearchOption searchOption) [0x00048] in C:\projects\radarr-usby1\src\NzbDrone.Common\Disk\DiskProviderBase.cs:177
  at NzbDrone.Core.MediaFiles.MovieFileMovingService.TransferFile (NzbDrone.Core.MediaFiles.MovieFile movieFile, NzbDrone.Core.Tv.Movie movie, System.String destinationFilePath, NzbDrone.Common.Disk.TransferMode mode) [0x00219] in C:\projects\radarr-usby1\src\NzbDrone.Core\MediaFiles\MovieFileMovingService.cs:147
  at NzbDrone.Core.MediaFiles.MovieFileMovingService.MoveMovieFile (NzbDrone.Core.MediaFiles.MovieFile movieFile, NzbDrone.Core.Parser.Model.LocalMovie localMovie) [0x00052] in C:\projects\radarr-usby1\src\NzbDrone.Core\MediaFiles\MovieFileMovingService.cs:82
  at NzbDrone.Core.MediaFiles.UpgradeMediaFileService.UpgradeMovieFile (NzbDrone.Core.MediaFiles.MovieFile movieFile, NzbDrone.Core.Parser.Model.LocalMovie localMovie, System.Boolean copyOnly) [0x000b0] in C:\projects\radarr-usby1\src\NzbDrone.Core\MediaFiles\UpgradeMediaFileService.cs:73
  at NzbDrone.Core.MediaFiles.EpisodeImport.ImportApprovedMovie.Import (System.Collections.Generic.List`1[T] decisions, System.Boolean newDownload, NzbDrone.Core.Download.DownloadClientItem downloadClientItem, NzbDrone.Core.MediaFiles.EpisodeImport.ImportMode importMode) [0x0026b] in C:\projects\radarr-usby1\src\NzbDrone.Core\MediaFiles\EpisodeImport\ImportApprovedMovie.cs:110
```

EDIT: Actually this could be related to #2485.

**Comment by speedyflipper** [comment]
I am having the same issue but am currently running the most recent update.

The movie will download and process correctly into the appropriate folder, however Radarr keeps the movie on the activity page. In addition no files are listed under the movie and the scan fails to find the files in the folder.

I did notice that the folder path listed is not correct. For example with Clue the correct folder is "Movies\Clue (1985) - tt0088930" (which is where the file is processed to) however path listed on the movie page is "Movies\Clue (1985) - ".

One workaround I found is to remove the movie from Radarr without deleting the files and reimport the file but then I lose the history which I like to keep
